server/index.js

```diff
@@ -1,36 +1,32 @@
-const app = new (require('koa'))()
+const Koa = require('koa'),
+  websockify = require('koa-websocket');
+
+const app = websockify(new Koa());
+let todos = new (require('immutable')).Map()
 
 app.use(require('koa-body')())
 
-let todos = new (require('immutable')).Map()
-const RE = new RegExp('/todos(?:/(.+))?')
-app.use(async (ctx, next) => {
-  const matches = RE.exec(ctx.url)
-  if (matches === null) {
-    return next()
-  }
+app.ws.use((ctx, next) =>{
+    ctx.websocket.send(JSON.stringify(todos));
 
-  const id = matches[1]
-  const { method } = ctx
-  if (method === 'GET') {
-    if (id === undefined) {
-      ctx.body = todos.toArray()
-    } else {
-      ctx.body = todos.get(id)
-    }
-  } else if (method === 'PUT' && id !== undefined) {
-    const todo = {
-      ...ctx.request.body,
-      id
-    }
-    todos = todos.set(id, todo)
-    ctx.body = 'ok'
-  }
-})
+    ctx.websocket.on('message', function (data) {
+      let msg = JSON.parse(data);
+      if(msg.action === "delete"){
+        const { id } = msg
+        todos = todos.delete(id)
+        ctx.websocket.send(JSON.stringify(todos));
+      } else if(msg.action === "set"){
+        let { todo } = msg
+        let { id } = todo
+        todos = todos.set(id, todo)
+        ctx.websocket.send(JSON.stringify(todos));
+      }
+  });
+});
 
-app.use(require('koa-static')(`${__dirname}/../pages/build`))
+app.use(require('koa-static')(`${__dirname}/../pages/build`));
 
-app.listen(20431, function (error) {
+app.listen(3000, function (error) {
```

Comment:
Why did you change the port?

Reply:
Just for better remembering the port number.